### PR TITLE
feat: Add CLI `register-oracle`

### LINKED ISSUE #2: Add CLI: register-oracle
Should be done with https://github.com/medibloc/panacea-core/issues/509

- Execute RegisterOracle tx to Panacea

### PR DESCRIPTION
**Pull request by H4NLee**
close #2

### CODE REVIEW

File: cmd/oracled/cmd/register_oracle.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			}
+
+			err = <-errChan
+			if err != nil {
+				log.Infof("Error occurs while getting shared oracle private key. Please retrieve it via get-oracle-key cmd: %v", err)
+				return err
+			} else {
+				log.Infof("oracle private key is successfully shared. You can start oracle now!")
+				return nil
+			}
```
Comment (H4NLee) [inline cmd/oracled/cmd/register_oracle.go:119]: If `MsgRegisterOracle` succeed, subscribe `MsgApproveOracleRegistration` and handle it in a synchronous way.

**Review by audtlr24 — APPROVED** [review]
Thanks 👍 lgtm!
But I have one question here.

File: cmd/oracled/cmd/register_oracle.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+			errChan := make(chan error, 1)
+
+			for tx := range txs {
+				errChan <- event.EventHandler(tx)
+			}
+
+			err = <-errChan
+			if err != nil {
+				log.Infof("Error occurs while getting shared oracle private key. Please retrieve it via get-oracle-key cmd: %v", err)
```
[the remark this answers is not in the document]
Comment (H4NLee) [reply cmd/oracled/cmd/register_oracle.go:114]: Right.

File: cmd/oracled/cmd/register_oracle.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			client, err := rpchttp.New(conf.Panacea.RPCAddr, "/websocket")
+			if err != nil {
+				return err
+			}
+
+			if err := client.Start(); err != nil {
+				return err
+			}
+			defer func() {
+				_ = client.Stop()
```
Comment (gyuguen) [inline cmd/oracled/cmd/register_oracle.go:93]: How about printing a log when error occurs?

File: cmd/oracled/cmd/register_oracle.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+			txs, err := client.Subscribe(ctx, "", event.GetEventQuery())
+			if err != nil {
+				return err
+			}
+
+			errChan := make(chan error, 1)
+
+			for tx := range txs {
+				errChan <- event.EventHandler(tx)
+			}
```
Comment (gyuguen) [inline cmd/oracled/cmd/register_oracle.go:110]: I'm not sure if this code runs correctly.

If registering a two new oracles, maybe occurs same tx.
Let's check it out in the actual implementation part

**Review by gyuguen — APPROVED** [review]
Good

File: cmd/oracled/cmd/register_oracle.go (hunk 4 above)
Comment (H4NLee) [reply cmd/oracled/cmd/register_oracle.go:110]: Could you elaborate a bit more what do you mean of registering two new oracles?

File: cmd/oracled/cmd/register_oracle.go (hunk 3 above)
Comment (H4NLee) [reply cmd/oracled/cmd/register_oracle.go:93]: I'll add it. Thx!

File: event/oracle/approve_oracle_registration_event.go [hunk 5]
```diff
@@ -8,18 +8,17 @@ import (
 var _ event.Event = (*ApproveOracleRegistrationEvent)(nil)
 
 type ApproveOracleRegistrationEvent struct {
-	reactor event.Reactor
 }
 
-func NewApproveOracleRegistrationEvent(s event.Reactor) ApproveOracleRegistrationEvent {
-	return ApproveOracleRegistrationEvent{s}
+func NewApproveOracleRegistrationEvent() ApproveOracleRegistrationEvent {
```
Comment (inchori) [inline event/oracle/approve_oracle_registration_event.go:13]: I have a question. Why remove the reactor for now?

**Review by inchori — APPROVED** [review]
LGTM 👍

File: event/oracle/approve_oracle_registration_event.go (hunk 5 above)
Comment (H4NLee) [reply event/oracle/approve_oracle_registration_event.go:13]: For this PR, I don't want to create a service for reactor, which has many things in it.
This `ApproveOracleRegistrationEvent` don't need all that resources in service.
I've been thinking about how to make this `ApproveOracleRegistrationEvent` better. If you have any idea, please feel free to leave your thoughts.

Comment (inchori) [reply event/oracle/approve_oracle_registration_event.go:13]: I understand it! I'll also thinking about how to make it better. Thanks!

**Review by youngjoon-lee — APPROVED** [review]
good

**Pull request merged**